### ISSUE TITLE
Differences in sr25519 implementations in JS and rust

### ISSUE THREAD

**Issue by ameba23** [entropyxyz/x25519-chacha20poly1305#15]
This relates to this crate using schnorrkel's [SecretKey::from_ed25519_bytes](https://docs.rs/schnorrkel/latest/schnorrkel/keys/struct.SecretKey.html#method.from_ed25519_bytes) when creating a sr25519 keypair from a given secret key.

It seems this is needed due to the different way sr25519 keypairs are created in [@poltadot/keyring](https://github.com/polkadot-js/common/tree/master/packages/keyring) and [sp-core::sr25519](https://docs.rs/sp-core/latest/sp_core/sr25519/index.html).

See the discussion in this PR:
https://github.com/entropyxyz/x25519-chacha20poly1305/pull/12

@frankiebee found this open issue in the schnorrkel repo where someone else has found this discrepancy: https://github.com/w3f/schnorrkel/issues/77

@frankiebee also found where this `from_ed25519_bytes` gets used in polkadot's `wasm-crypto` which is called internally when sr25519 signing using `@polkadot/keyring`:
https://github.com/polkadot-js/wasm/blob/b82e28ad4ed888ef4a461ee8e0ba25acdb87dc1a/packages/wasm-crypto/src/rs/sr25519.rs#L125

TLDR:  sr25519 Signing is **not the same** in polkadot-js and either of the substrate rust implementations (`sp-core::sr25519` and `subxt-signer`). The public key included in a `SignedMessage` created with `encrypt_and_sign` is different to the corresponding public key when creating a keypair in any sr25519 implementation, or with `SignedMessage::new`
